test/dnf/test_product_id.py

```diff
@@ -0,0 +1,97 @@
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# This software is licensed to you under the GNU General Public License,
+# version 2 (GPLv2). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+# along with this software; if not, see
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+#
+# Red Hat trademarks are not licensed under GPLv2. No permission is
+# granted to use or replicate Red Hat trademarks that are incorporated
+# in this software or its documentation.
+#
+
+import unittest
+import os
+import tempfile
+from typing import Union
+
+from plugins.dnf.product_id import DnfProductManager
```

Comment:
before importing the module, better try to import the dnf modules for it, so it is skipped on systems without dnf; see
https://github.com/candlepin/subscription-manager/blob/b31cb8aa032f37ccb236d4b693e057f79a659389/test/test_dnf_content_plugin.py#L7-L11

Reply:
Good point.